dataactbroker/helpers/fabs_derivations_helper.py

```diff
@@ -170,6 +170,14 @@ def derive_ppop_location_data(obj, sess, ppop_code, ppop_state_code, county_dict
         # deriving PrimaryPlaceOfPerformanceCityName
         city_info = sess.query(ZipCity).filter_by(zip_code=zip_five).one()
         obj['place_of_performance_city'] = city_info.city_name
+
+        # deriving PrimaryPlaceOfPerformanceScope
+        if ppop_code and (re.match('^[A-Z]{2}\d{5}$', ppop_code) or re.match('^[A-Z]{2}\d{4}R$', ppop_code)):
+            obj['place_of_performance_scope'] = "Single ZIP Code"
+    # if there is zip4 and it *is* city-wide, set the scope
+    elif obj['place_of_performance_zip4a'] and \
```

Comment:
Why don't we check for the existence of `ppop_code` here? We do in the above and it's just as likely to not exist here as it is up there. In fact, we could put this inside the `elif ppop_code` since that's required for this check anyway